src/errors.ts

```diff
@@ -0,0 +1,5 @@
+/**
+ * when remote pinning service returns delegates, if we can't connect to any, we won't be able to provide our CID's
+ * content to the service, and must abort.
+ */
+export class FailedToConnectToDelegates extends Error {}
```

Comment:
Without adding a code or other context does this class give us anything?

Reply:
just the name and ability to extend. we should have proper error codes, i will add them

Reply:
it also allows consumers to `instanceOf` check error types and handle appropriately.

Reply:
added explicit name in https://github.com/ipfs/helia-remote-pinning/pull/2/commits/5fe0bd98cbb3471e1e29bc1e756dd4a7a39845c2